### ISSUE TITLE
False positive E1101 accessing class attribute

### ISSUE THREAD

**Issue by hemberger** [PyCQA/pylint#2735]
### Steps to reproduce
```python
class Test():
    def __init__(self):
        self.x = 'I exist'
        [i.x for i in [self] + [self]]
```

### Current behavior
E1101: Instance of 'list' has no 'x' member (no-member)

### Expected behavior
No error

### pylint --version output
pylint 2.2.2
astroid 2.1.0
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]

### Additional information
I should note that I've tried a few other variations of the offending line:
```python
        y = [self] + [self]
        [i.x for i in y]            # error
        [i.x for i in [] + [self]]  # error
        [i.x for i in [self]]       # no error
        [i.x for i in []]           # no error
        [i.x for i in [] + []]      # no error
```
So it seems to be related to the result of list addition.

Thanks for your work on pylint! It is a very valuable tool.

**Comment by hemberger** [comment]
This appears to be fixed in the development version.

pylint 2.3.0-dev1
astroid 2.2.0-dev
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
